R/score.R

```diff
@@ -110,53 +97,114 @@
 #' PLoS Comput Biol 15(2): e1006785. \doi{10.1371/journal.pcbi.1006785}
 #' @export
 
-score <- function(data,
-                  metrics = NULL,
-                  ...) {
+score <- function(data, ...) {
+  UseMethod("score")
+}
+
+#' @rdname score
+#' @export
+score.default <- function(data, ...) {
+  data <- validate(data)
+  score(data, ...)
+}
 
-  # preparations ---------------------------------------------------------------
-  if (is_scoringutils_check(data)) {
-    check_data <- data
-  } else {
-    check_data <- check_forecasts(data)
-  }
+#' @rdname score
+#' @export
+score.scoringutils_binary <- function(data, metrics = metrics_binary, ...) {
+  data <- validate(data)
```

Comment:
would this catch if the input wasn't binary (and so on for the other functions)?

Reply:
If you somehow ended up with an object of class `scoringutils_binary` that doesn't conform to that standard, then it would error in `validate.scoringutils_binary()`